data/postgres/01_loads.sql

```diff
@@ -1,205 +1,262 @@
 CREATE SCHEMA IF NOT EXISTS loads;
 
-
--- The various sea states (to be expanded)
-DROP TYPE IF EXISTS sea_state CASCADE;
-CREATE TYPE sea_state AS ENUM ('wet');
-
 DROP TYPE IF EXISTS unit CASCADE;
-CREATE TYPE unit AS ENUM ('tonne', 'meter', 'knot', 'percentage');
-
-DROP TYPE IF EXISTS pcs_mode CASCADE;
-CREATE TYPE pcs_mode AS ENUM ('idle', 'regeneration', 'propulsion');
+CREATE TYPE unit AS ENUM ('tonne', 'percent', 'on-off');
 
-DROP TABLE IF EXISTS loads.masts CASCADE;
-CREATE TABLE loads.masts (
-  id TEXT PRIMARY KEY,
-  name TEXT
-);
+DROP TYPE IF EXISTS sail_position CASCADE;
+CREATE TYPE sail_position AS ENUM ('main', 'mizzen', 'fore-inner', 'fore-outer', 'mizzen-fore');
 
--- sails also includes different reefs of the main and mizzen sails. i.e. full-main-sail, main-sail-reef1, main-sail-reef2 are all rows in this table
 DROP TABLE IF EXISTS loads.sails CASCADE;
 CREATE TABLE loads.sails (
-  id TEXT PRIMARY KEY,
-  mast_id TEXT NOT NULL REFERENCES loads.masts(id) ON DELETE RESTRICT,
-  name TEXT
+    id TEXT PRIMARY KEY,             
+    abbreviation TEXT NOT NULL,               
+    position sail_position NOT NULL,       
+    name TEXT NOT NULL              
 );
 
 DROP TABLE IF EXISTS loads.sail_sets CASCADE;
 CREATE TABLE loads.sail_sets (
-  id TEXT PRIMARY KEY,
-  name TEXT NOT NULL
+    id SERIAL PRIMARY KEY,
+    main_sail_id TEXT REFERENCES loads.sails(id),
+    mizzen_sail_id TEXT REFERENCES loads.sails(id), 
+    fore_inner_sail_id TEXT REFERENCES loads.sails(id),
+    fore_outer_sail_id TEXT REFERENCES loads.sails(id),
+    mizzen_fore_sail_id TEXT REFERENCES loads.sails(id),
+    sail_set TEXT[] GENERATED ALWAYS AS (
+        ARRAY_REMOVE(ARRAY[main_sail_id, mizzen_sail_id, fore_inner_sail_id, fore_outer_sail_id, mizzen_fore_sail_id], NULL)
+    ) STORED,
+    UNIQUE(main_sail_id, mizzen_sail_id, fore_inner_sail_id, fore_outer_sail_id, mizzen_fore_sail_id)
```

Comment:
Welcome to SQL. Here `NULL` is never equal. Since unicity depends on equality, duplicates with `NULL`s are considered unique. The way to enforce uniqueness with `NULL`s is either with a filter on `IS NOT NULL`, but with 5 columns that gets complicated. You might instead use `COALESCE` with some "null string"